src/components/property-tree/property-tree.tsx

```diff
@@ -23,16 +23,18 @@ export const PropertyTree = ({ asset, childAssets }: PropertyTreeProps): JSX.Ele
   const excludedTreeAssets = "656feda1-896f-b136-da84-163ee4f1be6c"; // Hackney Homes
 
   const [treeViewData, setTreeViewData] = useState<TreeAsset[]>([]);
-  const childNodes = addChildrenAssets(childAssets);
-  const principle = generatePrinciple(asset, childNodes);
+  const [childNodes, setChildNodes] = useState<TreeAsset[]>([]);
 
   useEffect(() => {
+    setChildNodes(addChildrenAssets(childAssets));
+  }, [childAssets]);
+
+  useEffect(() => {
+    const principle = generatePrinciple(asset, childNodes);
     setTreeViewData(
       addParentsAndPrinciple(asset, childNodes, excludedTreeAssets, principle),
     );
-
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
```

Comment:
Im surprised the linter passes after removing that line. I would expect it to fail because you're not including the functions in the array

Reply:
It passes now because both useEffects have dependencies in their deps arrays. But yeah if you leave the array empty, that line is needed